### PR TITLE
Add API for creating filter conditions compatible with unique values selector

### PR DESCRIPTION
**Pull request by saskliutas**
Closes https://github.com/iTwin/presentation/issues/477

### CODE REVIEW

File: .changeset/metal-tables-wonder.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@itwin/presentation-components": patch
```
Comment (grigasp) [inline .changeset/metal-tables-wonder.md:2]: shouldn't this be a minor?

File: packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilter.ts [hunk 2]
```diff
@@ -529,3 +546,20 @@ function isPoint3dLike(value: object): value is { x: number; y: number; z: numbe
 function isInstanceKeyLike(value: object): value is { id: string; className: string } {
   return (value as any).id !== undefined && (value as any).className !== undefined;
 }
+
+function createUniqueValue(value: PropertyValue): PrimitiveValue | undefined {
+  if (value.valueFormat !== PropertyValueFormat.Primitive || value.displayValue === undefined || value.value === undefined) {
+    return undefined;
```
Comment (grigasp) [inline packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilter.ts:552]: doesn't this mean that we were given an invalid `PrimitiveValue`? maybe we should throw instead?

Comment (saskliutas) [reply packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilter.ts:552]: I guess we could throw if `valueFormat` is not `Primitive`. But as this is an utility function for creating conditions using unique values selector, I think makes it more convenient for consumer not throwing error.

File: packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilter.ts [hunk 3]
```diff
@@ -547,10 +555,11 @@ function isInstanceKeyLike(value: object): value is { id: string; className: str
   return (value as any).id !== undefined && (value as any).className !== undefined;
 }
 
-function createUniqueValue(value: PropertyValue): PrimitiveValue | undefined {
-  if (value.valueFormat !== PropertyValueFormat.Primitive || value.displayValue === undefined || value.value === undefined) {
-    return undefined;
-  }
+function isValidPrimitiveValue(val: PrimitiveValue): val is Required<PrimitiveValue> {
+  return val.valueFormat === PropertyValueFormat.Primitive && val.value !== undefined && val.displayValue !== undefined;
```
Comment (grigasp) [inline packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilter.ts:559]: is `val.valueFormat === PropertyValueFormat.Primitive` check necessary anymore?

File: packages/components/src/test/instance-filter-builder/PresentationInstanceFilter.test.ts [hunk 4]
```diff
@@ -985,27 +985,34 @@ describe("PresentationInstanceFilter", () => {
     });
   });
 
-  describe("createEqualityCondition", () => {
+  describe("createPrimiiveValueEqualityCondition", () => {
```
Comment (grigasp) [inline packages/components/src/test/instance-filter-builder/PresentationInstanceFilter.test.ts:988]:
```suggestion
  describe("createPrimitiveValueEqualityCondition", () => {
```

File: .changeset/metal-tables-wonder.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@itwin/presentation-components": minor
+---
+
+Added `PresentationInstanceFilter.createEqualityCondition` function to provide a convenient way for creating equality filter conditions compatible with `UniquePropertyValuesSelector`.
```
Comment (grigasp) [inline .changeset/metal-tables-wonder.md:5]:
```suggestion
Added `PresentationInstanceFilter.createPrimitiveValueEqualityCondition` function to provide a convenient way for creating equality filter conditions compatible with `UniquePropertyValuesSelector`.
```

**Review by grigasp — APPROVED** [review]

**Pull request merged**